conda_build/build.py

```diff
@@ -1102,6 +1102,53 @@ def sanitize_channel(channel):
     return get_conda_channel(channel).urls(with_credentials=False, subdirs=[''])[0]
 
 
+def get_git_info(m):
+    """
+    Grabs git info for a build
+    or
+    Raises
+    """
+    recipe_dir = m.path
+    command = ['git', '-C', recipe_dir, 'rev-parse', '--abbrev-ref', 'HEAD']
+    p = subprocess.Popen(
+            command,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            )
+    out, _ = p.communicate()
+    branch = out.decode("utf-8")[:-1]
+    if not branch:
+        raise
+
+    command = ['git', '-C', recipe_dir, 'rev-parse', f'origin/{branch}']
+    p = subprocess.Popen(
+            command,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            )
+    out, _ = p.communicate()
+    remote_commit = out.decode("utf-8")[:-1]
+
+    command = ['git', '-C', recipe_dir, 'remote', 'get-url', 'origin']
+    p = subprocess.Popen(
+            command,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            )
+    out, _ = p.communicate()
+    origin_url = out.decode("utf-8")[:-1]
+
+    git_info = {
+            "origin_url": origin_url,
+            "commit": remote_commit
+            }
+
+    return git_info
```

Comment:
This all should use [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run) instead to simplify the code and improve error handling.